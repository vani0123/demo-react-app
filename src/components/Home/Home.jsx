import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { PropTypes } from 'prop-types';
import { withTranslation } from 'react-i18next';
import { sessionAction } from '../../redux/actions';
import { APP_CONSTANTS, I18N_CONSTANTS } from '../../constants';
import { WelcomePage } from '../Common';

class Home extends Component {
  constructor(props, context) {
    super(props, context);

    this.onThemeChange = this.onThemeChange.bind(this);
  }

  componentDidMount() {}

  onThemeChange(event) {
    const { actions } = this.props;

    const theme = event.target.checked
      ? APP_CONSTANTS.THEME.DARK
      : APP_CONSTANTS.THEME.DEFAULT;

    actions.currentTheme(theme);
  }

  render() {
    const { session, t } = this.props;
    return (
      <div className="metlife">
        <WelcomePage
          onThemeChange={this.onThemeChange}
          currentTheme={session.currentTheme}
          t={t}
        />
      </div>
    );
  }
}

Home.propTypes = {
  session: PropTypes.shape({
    currentTheme: PropTypes.string.isRequired
  }).isRequired,
  actions: PropTypes.shape({
    currentTheme: PropTypes.func.isRequired
  }).isRequired,
  t: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    session: state.session
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...sessionAction }, dispatch)
  };
}

const component = connect(mapStateToProps, mapDispatchToProps)(Home);
export default withTranslation(I18N_CONSTANTS.NAMESPACE.HOME)(component);
