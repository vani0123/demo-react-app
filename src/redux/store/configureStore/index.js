import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import createSagaMiddleware from 'redux-saga';
import rootReducer from '../../reducers';
import { initSagas } from '../../sagas/initSagas';

export default function configureStore(initialState) {
  const shouldHideDevTool = process.env.NODE_ENV === 'production'
    || typeof document === 'undefined'
    || !window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;

  const composeEnhancers = shouldHideDevTool
    ? compose
    : window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__; // add support for Redux dev tools

  const sagaMiddleware = createSagaMiddleware({
    onError(error) {
      setImmediate(() => {
        throw error;
      });
    }
  });
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(sagaMiddleware, reduxImmutableStateInvariant())
    )
  );

  initSagas(sagaMiddleware);

  return store;
}
