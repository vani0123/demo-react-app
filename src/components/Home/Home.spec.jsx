import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../../redux/store/configureStore';
import Home from './Home';

const setup = (props = {}) => {
  const store = configureStore();
  const defaultProps = {
    isLoading: false,
    ...props
  };

  return mount(
    <Router>
      <Provider store={store}>
        <Home {...defaultProps} />
      </Provider>
    </Router>
  );
};

describe('Home component', () => {
  it('should renders without crashing', () => {
    const props = { t: jest.fn((x) => x) };
    const wrapper = setup(props);
    const { isLoading } = wrapper.find('Home').props();

    expect(isLoading).toBeFalsy();
  });

  it('should have valid snapshot', () => {
    const props = { t: jest.fn((x) => x) };
    const wrapper = setup(props);

    expect(wrapper).toMatchSnapshot();
  });
});
