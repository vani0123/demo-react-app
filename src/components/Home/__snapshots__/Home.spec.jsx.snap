// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should have valid snapshot 1`] = `ReactWrapper {}`;
